ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -315,7 +324,7 @@ private void unregisterQuery(final QueryMetadata query) {
       }
     }
 
-    allLiveQueries.remove(query);
+    allLiveQueries.remove(query.getQueryId(), query);
```

Comment:
It's probably correct to do it this way since we always expect query to be the existing object, but it's probably less confusing to simplify this to just `allLiveQueries.remove(query.getQueryId())` so that a reader wouldn't think query might not match in some circumstances.